src/main/java/com/iota/iri/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java

```diff
@@ -154,6 +173,39 @@ private void latestSolidMilestoneTrackerThread() {
         }
     }
 
+    /**
+     * Applies the given milestone to the ledger.<br />
+     * <br />
+     * If the application of the milestone fails, we start a repair routine which will revert the milestones preceding
+     * our current milestone and consequently try to reapply them in the next iteration of the {@link
+     * #checkForNewLatestSolidMilestones()} method (until the problem is solved).<br />
+     *
+     * @param milestone the milestone that shall be applied to the ledger state
+     * @throws Exception if anything unexpected goes wrong while applying the milestone to the ledger
+     */
+    private void applySolidMilestoneToLedger(MilestoneViewModel milestone) throws Exception {
+        if (!ledgerService.applyMilestoneToLedger(milestone)) {
```

Comment:
this reverse if statement took me a while to read - better to have the positive (common case) at the beginning IMO.
(i.e. swap)

Reply:
changed it and introduces some additional utility methods for better readability